CS/feature Comprehensive Event Filtering — Integrated Location Services, Location Normalization, and Multi-Factor Filtering

cesarsanchez1 opened the pull request:
### **Ready for Review**

This PR adds the full event filtering system we’ve been working toward. The filters objective is to make the event list actually useful by letting people narrow things down by location, date, ticket type, price, etc., and everything now plays nicely together.

**Here’s what’s included:**

- Location Features
- Added sorting by distance (“Nearest First” / “Farthest First”).
- Added radius filtering (5–100 miles).
- Uses the Haversine formula to compute distance in SQL.
- **Supports on-demand automatic location detection only (no manual input), triggered when the user selects distance-based filters.**
- If the user already gave location permission, we reuse it (otherwise we request it only when needed).

**Sorting Updates**

- Sorting now works in layers:
- Date (if selected) always takes priority.
- Price sorting comes after that.
- Distance sorting comes last.
- Price sorting is based on General Admission ticket price.

**Filters**

- Multi-select state filter with a dropdown + “Select All”.
- Ticket type filter (General, Early Bird, VIP).
- Start/end date filtering with auto-submit.

**Template + UI**

- Updated event_list.html to add distance and radius controls.
- Cleaned up layout and spacing so the filters don’t fight each other.
- Added JS to handle geolocation, radius logic, and the select-all behavior.

**Backend Improvements**

- Distance is only calculated when needed, makes querying fast.
- Radius filter now works based on actual computed distance.
- All filters combine cleanly without breaking each other.

Commit:
feat: stabilize event filtering and price sorting (GA-based) with normalized state dropdown

Commit:
feat: redesigned event filter bar layout and added date picker with Flatpickr integration

Commit:
stable bootstrap version

Commit:
Add compact filter bar styling and integrate with futuristic event list UI

Commit:
Stable UI version: compact filter bar + multi-state dropdown + hybrid sorting

Commit:
Add distance sorting + radius filtering with geolocation; fix Haversine annotation; integrate with existing filters; rigorously tested stable version

Commit:
Remove State Filter UI from event_list.html (logic preserved)

Commented out the State filter section in the template based on reviewer feedback. Backend logic remains intact so the feature can easily be restored later without reimplementation.

Commit:
Enhance CI: add flake8 + black formatting checks before tests

thaddeuspearson commented on simpletix/simpletix/templates/simpletix/nav.html:
```diff
@@ -26,12 +27,40 @@
+              <span class="input-group-text bg-white border-end-0">
+                <i class="bi bi-geo-alt"></i>
+              </span>
+              <input id="location-input" class="form-control border-start-0" type="text" placeholder="Enter city or ZIP">
```
I'm a little confused as to what this is supposed to do for us. As it stands now, I seem to be able to enter a city manually, and it simply persists but no real filtering is happening with it.

For example, I typed in New York, and then used the distance filter (5 miles), but it used my location and only showed an event in Madrid.

Does this field serve a specific purpose?

thaddeuspearson commented on simpletix/simpletix/templates/simpletix/nav.html:
```diff
@@ -26,12 +27,40 @@
+              <input id="location-input" class="form-control border-start-0" type="text" placeholder="Enter city or ZIP">
+
+              <!-- Auto-detect location button -->
+              <button id="set-location-btn" class="btn btn-outline-light ms-2" title="Use my location">
```
I like this button feature, but if it is going to be used in conjunction with the location input field above, I feel like clicking it should auto-populate the field with the city it detects. Right now, it just simply gets the longitude/latitude and sets them as query_params.

If we are going to scrap the location input field altogether, then we should also scrap this, and simply let the filter field itself prompt the location when a user selects a distance (5 miles, 10 miles ...) automatically

thaddeuspearson reviewed (changes requested):
This is an excellent feature, but I have a concern about the location input field in the nav bar. Right now, it really isn't doing anything. I left a comment on the form directly for discussion.

Also, we need to add some test cases for the `event_list` view to make sure our coverage stays high.

Commit:
Cleanup: remove unused location UI from navbar

Scrapped manual location input and redundant auto-detect button as per reviewer feedback. Nav now only handles search UI; all geolocation logic centralized in event_list.

Commit:
Cleaned navbar: removed unused manual location UI for clarity

thaddeuspearson commented on simpletix/simpletix/templates/simpletix/nav.html:
```diff
@@ -24,75 +25,52 @@
-                {% if request.session.desired_role == 'organizer' %}
-                <li><a class="dropdown-item" href="{% url 'events:create_event' %}">Create Event</a></li>
-                {% endif %}
+                <li><a class="dropdown-item" href="/webpage/Find%20Event%20Page">Find Event</a></li>
```
@cesarsanchez1 Is there a reason why we switched from the dynamic links to hardcoding these values?

thaddeuspearson commented on simpletix/simpletix/templates/simpletix/nav.html:
```diff
@@ -24,75 +25,52 @@
               </ul>
             </li>
             <li class="nav-item">
-              <a class="nav-link text-white" href="{% url 'tickets:ticket_list' %}" role="button" aria-expanded="false">My Tickets</a>
```
and here?

thaddeuspearson commented on simpletix/simpletix/templates/simpletix/nav.html:
```diff
@@ -24,75 +25,52 @@
-                  {% endif %}
-                  <span class="ms-2">{{ user.username }}</span>
+                  
+                    <img src="https://simpletix-dev-media.s3.amazonaws.com/profile_photos/post-the-best-cat-memes-you-got-in-the-comments-please-v0-kg82lbnu0ste1.png.jpg?AWSAccessKeyId=ASIASAZS2CFM7PNSPBBW&amp;Signature=8E%2F2nXcL%2BcqBmdghk6fLFMdoNGw%3D&amp;x-amz-security-token=IQoJb3JpZ2luX2VjELL%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJHMEUCIEbFRjtmYP0AmIUjNDbrDhVpjqvKV%2BhoLvfFT%2Bd3yZ% ...
```
@cesarsanchez1 are we hard-coding your profile pic here?

thaddeuspearson commented on simpletix/simpletix/templates/simpletix/nav.html:
```diff
@@ -24,75 +25,52 @@
+                  
+                    <img src="https://simpletix-dev-media.s3.amazonaws.com/profile_photos/post-the-best-cat-memes-you-got-in-the-comments-please-v0-kg82lbnu0ste1.png.jpg?AWSAccessKeyId=ASIASAZS2CFM7PNSPBBW&amp;Signature=8E%2F2nXcL%2BcqBmdghk6fLFMdoNGw%3D&amp;x-amz-security-token=IQoJb3JpZ2luX2VjELL%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJHMEUCIEbFRjtmYP0AmIUjNDbrDhVpjqvKV%2BhoLvfFT%2Bd3yZ% ...
+                  
+                  <span class="ms-2">cesarsa7932</span>
```
@cesarsanchez1 and your username?

cesarsanchez1 replied:
I reverted the nav.html files to their original state so they now are unchanged in this pr. Apologies for the confusion I tried to manually undo the changes to nav and mistakenly added these. I'll drop a slack message when its ready to review

thaddeuspearson commented on simpletix/simpletix/templates/simpletix/nav.html:
```diff
@@ -24,75 +25,52 @@
+                  <span class="ms-2">cesarsa7932</span>
                 </a>
                 <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="userMenu">
-                  <li><a class="dropdown-item" href="{% url 'accounts:profile_edit' %}?next=/">Edit Profile</a></li>
```
same hardcoding url question

thaddeuspearson commented on simpletix/simpletix/templates/simpletix/nav.html:
```diff
@@ -24,75 +25,52 @@
-                  <li><a class="dropdown-item" href="{% url 'accounts:profile_edit' %}?next=/">Edit Profile</a></li>
+                  <li><a class="dropdown-item" href="/accounts/profile/edit/?next=/">Edit Profile</a></li>
                   <li><hr class="dropdown-divider"></li>
-                  <li><a class="dropdown-item" href="{% url 'accounts:logout' %}">Logout</a></li>
```
and here

cesarsanchez1 replied:
Oh my bad, this isnt ready for review yet, I made significant changes to the nav bar html and was trying to revert that file specifically without touching the other changes to the other files. This latest committed version isn't ready to review yet. Ill take care of it now

thaddeuspearson reviewed (changes requested):
Saw a few more things we need to make sure are addressed before merging this. Let me know if you have any questions

Commit:
Restore nav.html to dynamic template version

Removed accidental hard-coded values introduced during manual revert. Restored correct Django template logic.

Commit:
Restore original style.css before PR changes

Reverting to clean baseline before adding location-banner CSS.

Commit:
Remove state filters and restore original CSS file

Cleaned up event_list template, removed state filter UI, and commented backend state logic.

cesarsanchez1 commented:
@thaddeuspearson Hey, this is now ready for review. I resolved the lingering issues with this. I commented out all frontend and backend code related to the state-based filters which we decided we were omitting (we can delete them later if we decide to). I also restored the nav.html file, which previously had some mistakenly hardcoded details. A test file was also created and pasted test.

**Important note on geolocation:**
Both distance-based filters correctly prompt the user for location. However, modern browsers only allow the geolocation prompt to appear once per domain. If the user denies permission, the browser won’t reprompt, the user has to manually re-enable location access in their browser settings. I tested this in both Safari and Chrome: once location access is re-enabled manually, the location based filters work normally again.

Browsers have strict rules here, so there’s no reliable way around this from our side. The UI handles it by showing an alert when permission is missing, but we can’t override the browser’s built-in permission system.

Commit:
Add event distance filter tests and fix missing annotation logic

Includes new unit tests for distance sorting and radius behavior, plus flake8 fixes and safe ordering guard.

Commit:
Merge remote-tracking branch 'origin/develop' into cs/feature/localized-event-filtering

Commit:
Merge branch 'develop' of https://github.com/gcivil-nyu-org/team1-mon-fall25 into cs/feature/localized-event-filtering

Commit:
UI fixes: darken ticket-type labels and resolve darkened calendar input issue in event_list.html

Commit:
Merge branch 'develop' of https://github.com/gcivil-nyu-org/team1-mon-fall25 into cs/feature/localized-event-filtering

thaddeuspearson reviewed (approved):
It looks like the CSS issues for the drop down menus have been resolved! LGTM
